pkg/api/rest/delete.go

```diff
@@ -40,14 +45,17 @@ type RESTDeleteStrategy interface {
 // condition cannot be checked or the gracePeriodSeconds is invalid. The options argument may be updated with
 // default values if graceful is true.
 func BeforeDelete(strategy RESTDeleteStrategy, ctx api.Context, obj runtime.Object, options *api.DeleteOptions) (graceful, gracefulPending bool, err error) {
-	if strategy == nil {
-		return false, false, nil
-	}
-	objectMeta, _, kerr := objectMetaAndKind(strategy, obj)
+	objectMeta, gvk, kerr := objectMetaAndKind(strategy, obj)
 	if kerr != nil {
 		return false, false, kerr
 	}
-
+	if options.Preconditions.UID != nil && *options.Preconditions.UID != objectMeta.UID {
+		return false, false, errors.NewConflict(unversioned.GroupResource{gvk.Group, gvk.Kind}, objectMeta.Name, fmt.Errorf("the UID in the precondition (%s) does not match the UID in record (%s). The object might have been deleted and then recreated", *options.Preconditions.UID, objectMeta.UID))
```

Comment:
Add a comment to that effect, then.

On Thu, Mar 17, 2016 at 3:53 PM, Chao Xu <notifications@github.com> wrote:

> In pkg/api/rest/delete.go
> <https://github.com/kubernetes/kubernetes/pull/22965#discussion_r56591093>
> :
>
> >  	if kerr != nil {
> >  		return false, false, kerr
> >  	}
> > -
> > +	if options.Preconditions.UID != nil && *options.Preconditions.UID != objectMeta.UID {
> > +		return false, false, errors.NewConflict(unversioned.GroupResource{gvk.Group, gvk.Kind}, objectMeta.Name, fmt.Errorf("the UID in the precondition (%s) does not match the UID in record (%s). The object might have been deleted and then recreated", *options.Preconditions.UID, objectMeta.UID))
>
> The check here will help fail early, so I think we can keep them.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/22965/files/754caf176c23fe105cf47de2eb51aa89753e325a#r56591093>
>